indexlAlg.js

```diff
@@ -0,0 +1,621 @@
+
+// (function getMassiv() {
+
+//   for (let i = 1; i < 6; i++) {
+//     let str = '';
+//     for (let j = 1; j <= i; j++) {
+//       str = str.concat(i);
+//     }
+
+//     mas.push(str);
+//   }
+//   let k = console.log(mas);
+
+//   return k;
+// }())
+
+// 1. Fill. Напишите функцию, которая заполняет новый массив предоставленным значением.
+
+//   * Описание задачи: Напишите функцию, которая заполняет новый массив предоставленным значением.
+//   * Ожидаемый результат: (3, 'a') => ['a', 'a', 'a']
+//   * Сложность задачи: 1 of 5
+//   * @param {number} arraySize - размер массива
+//   * @param {?} value - значение для массива
+//   * @returns {Array}
+
+// const fillArray = (arraySize, value) => {
+//   let array = [];
+
+//   for (let i = 1; i <= arraySize; i++) {
+//     array.push(value);
+//   }
+
+//   return array
+// }
+
+// const data = 8;
+// const valueToFill = 'c';
+
+// console.log(fillArray(data, valueToFill)) // ['a', 'a', 'a']
+
+// 2. Reverse. Напишите функцию, которая разворачивает массив в обратном порядке. Пожалуйста, не используйте array.reverse(), чтобы сделать задачу более интересной.
+
+//   * Описание задачи: Напишите функцию, которая разворачивает массив в обратном порядке.
+//   * Ожидаемый результат: [1, 2, 3] => [3, 2, 1]
+//   * Сложность задачи: 1 of 5
+//   * @param {Array} array - Массив любых элементов
+//   * @returns {Array}
+
+// const reverse = (array) => {
+//   let newArray = [];
+
+//   for (let i = array.length - 1; i >= 0; i--) {
+//     newArray = newArray.concat([array[i]])
+//   }
+
+//   return newArray
+// }
+
+// const dat = [1, 2, 3];
+// console.log(reverse(dat)); // [3, 2, 1]
+
+
+// 3. Compact. Напишите функцию, которая очищает массив от нежелательных значений, таких как false, undefined, пустые строки, ноль, null.
+
+//  * Описание задачи: Напишите функцию, которая очищает массив от нежелательных значений, таких как false, undefined, пустые строки, ноль, null.
+//   * Ожидаемый результат: [0, 1, false, 2, undefined, '', 3, null] => [1, 2, 3]
+//   * Сложность задачи: 1 of 5
+//   * @param {Array} array - Массив любых элементов
+//   * @returns {Array}
+
+// const compact = (array) => {
+//   let newArray = array.filter(Boolean);
+//   return newArray;
+// }
+
+// const data = [0, 1, false, 2, undefined, '', 3, null];
+// console.log(compact(data)) // [1, 2, 3]
+
+
+// 4. From Pairs. Напишите функцию, которая возвращает объект, составленный из значений вложенных массивов. Первое элемент массива - ключ, второй - зачение. (используем метод массива reduce)
+
+//   * Описание задачи: Напишите функцию, которая возвращает объект, составленный из значений вложенных массивов. Первое значение - ключ, второе - зачение.
+//   * Ожидаемый результат: [['a', 1], ['b', 2]] => { a: 1, b: 2 }
+//   * Сложность задачи: 2 of 5
+//   * @param {Array} array - массив, значения которого массивы пар
+//   * @returns {Array}
+
+// let obj = {};
+
+// const fromPairs = (array) => array.reduce((prev, item) => {
```

Comment:
```suggestion
// const fromPairs = array => array.reduce((prev, item) => {
```

когда 1 параметр, скобки не нужны